src/tensorlake/applications/remote/runner.py

```diff
@@ -1,15 +1,38 @@
 from typing import Any
 
+import pydantic
+
 from ..function.user_data_serializer import serialize_value
 from ..interface.request import Request
 from ..metadata import ValueMetadata
-from ..user_data_serializer import UserDataSerializer, serializer_by_name
+from ..user_data_serializer import (
+    PickleUserDataSerializer,
+    UserDataSerializer,
+    serializer_by_name,
+)
 from .api_client import APIClient
 from .app_manifest_cache import get_app_manifest, has_app_manifest, set_app_manifest
 from .manifests.application import ApplicationManifest
 from .request import RemoteRequest
 
 
+def _convert_pydantic_to_dict(obj: Any) -> Any:
+    """Recursively converts Pydantic models to dicts.
+
+    This is used for pickle serialization to avoid class reference issues
+    when sending Pydantic instances across process boundaries.
+    """
+    if isinstance(obj, pydantic.BaseModel):
+        return obj.model_dump()
```

Comment:
We handle all special Pydantic scenarios in json use data serializer. We need to try to keep all such code in common places like json serializer to not create code duplications.